### PR TITLE
Noop DiagnosticsManager if HighFive is disabled

### PR DESCRIPTION
**Pull request by Dekken**
https://github.com/PHAREHUB/PHARE/issues/224

+ some Werror in ```src/python3/data_wrangler.cpp```

may be replaced by question. 

Is one Diagnostics format mandatory?

### CODE REVIEW

File: src/diagnostic/diagnostic_manager.h [hunk 1]
```diff
@@ -103,20 +100,24 @@ class DiagnosticsManager : public IDiagnosticsManager
         return diag.computeTimestamps[lastCompute] + timeStep > timeStamp;
     }
 
-
+    Writer& writer() { return *writer_.get(); }
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/diagnostic_manager.h:103]: can add constructor check sure

File: src/diagnostic/detail/h5writer.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ class Writer : public PHARE::diagnostic::IWriter
     static constexpr auto dimension   = GridLayout::dimension;
     static constexpr auto interpOrder = GridLayout::interp_order;
 
-    Writer(ModelView& modelView, std::string const hifivePath,
+    Writer(ModelView&& modelView, std::string const hifivePath,
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/detail/h5writer.h:41]: ModelView is just a wrapper for forwarding some internals, it's not all that complex.

File: src/diagnostic/detail/h5writer.h [hunk 3]
```diff
@@ -139,7 +141,7 @@ class Writer : public PHARE::diagnostic::IWriter
     double timestamp_ = 0;
     std::string filePath_;
     std::string patchPath_; // is passed around as "virtual write()" has no parameters
-    ModelView& modelView_;
+    ModelView modelView_;
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/detail/h5writer.h:144]: it needed by the class, so can be stored on the class.

File: src/diagnostic/diagnostic_manager.h [hunk 4]
```diff
@@ -2,6 +2,10 @@
 #ifndef PHARE_DIAGNOSTIC_MANAGER_HPP_
 #define PHARE_DIAGNOSTIC_MANAGER_HPP_
 
+#if !defined(PHARE_HAS_HIGHFIVE)
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/diagnostic_manager.h:5]: will move to "diagnostics.h" which is better

File: src/diagnostic/diagnostic_manager.h [hunk 5]
```diff
@@ -103,20 +100,24 @@ class DiagnosticsManager : public IDiagnosticsManager
...
-
+    Writer& writer() { return *writer_.get(); }
 
 protected:
     std::vector<DiagnosticProperties> diagnostics_;
 
 private:
-    Writer& writer_;
-
+    std::unique_ptr<Writer> writer_;
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/diagnostic_manager.h:109]: Writer has initialization from the PHAREDict, ModelView does not.

File: src/diagnostic/detail/h5writer.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/detail/h5writer.h:41]: could be a construct that takes a Hierarchy And HybridModel also.

[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/detail/h5writer.h]: > could be a construct that takes a Hierarchy And HybridModel also.

did this

File: src/diagnostic/diagnostic_manager.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/diagnostic_manager.h]: done

File: src/diagnostic/detail/h5writer.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicolasaunai) [reply src/diagnostic/detail/h5writer.h:145]: maybe silly question but: why asking for client to build it ? In the end, if it's only getting here through an rvalue, it probably means the caller will not have a copy/ref of it itself. Thus, why isn't the Writer building it itself as an internal ?

[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/detail/h5writer.h:145]: this is already the case, see the latest commit!

[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/detail/h5writer.h:145]: https://github.com/PHAREHUB/PHARE/pull/225/files#diff-e911b17b7ec740ea725c59750f6603dcR42

**Pull request merged**